via/static/scripts/video_player/components/test/ToastMessages-test.js

```diff
@@ -0,0 +1,109 @@
+import { mount } from 'enzyme';
+
+import { delay } from '../../../test-util/wait';
+import ToastMessages from '../ToastMessages';
+
+describe('ToastMessages', () => {
+  const toastMessages = [
+    {
+      id: 'abc',
+      type: 'success',
+      message: 'Hello world',
+    },
+    {
+      id: 'def',
+      type: 'success',
+      message: 'Foobar',
+    },
+    {
+      id: 'ghi',
+      type: 'error',
+      message: 'Something failed',
+    },
+  ];
+  let fakeOnMessageDismiss;
+
+  beforeEach(() => {
+    fakeOnMessageDismiss = sinon.stub();
+  });
+
+  function createToastMessages(toastMessages, setTimeout) {
+    const container = document.createElement('div');
+    document.body.appendChild(container);
+
+    return mount(
+      <ToastMessages
+        messages={toastMessages}
+        onMessageDismiss={fakeOnMessageDismiss}
+        _setTimeout={setTimeout}
+      />,
+      { attachTo: container }
+    );
+  }
+
+  it('renders a list of toast messages', () => {
+    const wrapper = createToastMessages(toastMessages);
+    assert.equal(wrapper.find('ToastMessageItem').length, toastMessages.length);
+  });
+
+  toastMessages.forEach((message, index) => {
+    it('dismisses messages when clicked', async () => {
+      const wrapper = createToastMessages(toastMessages);
+
+      wrapper.find('Callout').at(index).props().onClick();
+      // onMessageDismiss is not immediately called. Transition has to finish
+      assert.notCalled(fakeOnMessageDismiss);
+
+      // Once dismiss transition has finished, onMessageDismiss is called
+      await delay(350); // "out" transition lasts 300ms
+      assert.calledWith(fakeOnMessageDismiss, message.id);
+    });
+  });
+
+  it('dismisses messages automatically unless instructed otherwise', async () => {
+    createToastMessages(
+      [
+        ...toastMessages,
+        {
+          id: 'foo',
+          type: 'success',
+          message: 'Not to be dismissed',
+          autoDismiss: false,
+        },
+      ],
+      // Fake internal setTimeout, to immediately call its callback
```

Comment:
As a rule of thumb, I would recommend that fakes of async events should still be async, so as to better reflect the real flow of execution. Changing a fake timer from 100ms to 0ms makes the test run faster without affecting the overall flow of events.

Reply:
I'm not sure if I understand this 🤔 

Were you just making a general comment, or are you suggesting some change?

Reply:
This is just a general comment. I don't think it is critical to act on it here (but if you want to, that's fine too).